pandas/sparse/array.py

```diff
@@ -191,20 +189,18 @@ def __new__(cls, data, sparse_index=None, index=None, kind='integer',
         if dtype is not None:
             dtype = np.dtype(dtype)
         is_sparse_array = isinstance(data, SparseArray)
-        if fill_value is None:
-            if is_sparse_array:
-                fill_value = data.fill_value
-            else:
-                fill_value = nan
 
         if is_sparse_array:
             sparse_index = data.sp_index
-            values = np.asarray(data)
+            values = data.sp_values
+            fill_value = data.fill_value
         else:
             # array-like
             if sparse_index is None:
-                values, sparse_index = make_sparse(data, kind=kind,
-                                                   fill_value=fill_value)
+                if dtype is not None:
```

Comment:
actually maybe should put this in ``sparse.array`` or have a ``sparse.common`` module that you can import to do centrailzed sparse things .